<a href="https://colab.research.google.com/github/Teninlanimi/flyrank-interenship-ml/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

My Lane is the Refresh / Content Opportunity Scoring because it helps us make informed decisions on the so to say upkeep of our webpages.
Organizations often have hundreds or thousands of content pages but limited time and resources to review them all. A data-driven scoring system can help prioritize which pages should be refreshed, expanded, protected, monitored, or pruned based on observable performance signals. This project aims to support better decision-making by identifying pages that are most likely to benefit from editorial attention.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

1.What decision does my work improve
My work improves the decision of which content pages should be prioritized for action. Instead of manually reviewing hundreds or thousands of pages, the model ranks pages based on their likelihood of benefiting from actions such as refreshing, expanding, protecting, pruning, or monitoring. This helps ensure that the content team's time and resources are focused on pages with the highest potential impact.

2. Who acts on it?
SEO specialists, content strategists, editors, and marketing managers.

3.Cost of a wrong recommendation: Wasted editorial effort, missed SEO growth opportunities, potential traffic and revenue loss, and reduced confidence in the recommendation system. The model is intended to support not replace human decision making.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [13]:
import os, sys, subprocess

import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print(df.head())
print(df.shape)
print(df.columns.tolist())

             content_id          client_id  search_volume  competition  \
0  content_304f48230142  client_f369cb89fc           10.0         0.67   
1  content_a1fb4e703a9e  client_4e07408562           90.0         0.01   
2  content_9aa793d4d895  client_7f2253d7e2            0.0         0.00   
3  content_331d6c4de07b  client_19581e27de           10.0         0.00   
4  content_d99b7a2d90ca  client_3fdba35f04            0.0         0.00   

  competition_level   cpc     content_type    main_intent  word_count  \
0              HIGH  2.05  keyword article  transactional      3221.0   
1               LOW  0.05  keyword article  informational      2481.0   
2               LOW  0.00  keyword article  informational      3515.0   
3               LOW  0.00  keyword article     commercial         NaN   
4               LOW  0.00  keyword article  informational      2803.0   

   char_count  ... char_count_tier   ctr  avg_position  engagement_rate  \
0     20457.0  ...     15000-25000  0.76 

In [18]:
print("Number of content pages:", len(df))

print("\nTrend direction counts:")

counts = df["trend_direction"].value_counts()
percentages = df["trend_direction"].value_counts(normalize=True) * 100

summary = pd.DataFrame({
    "Count": counts,
    "Percentage": percentages.round(2)
})
print(summary)

print("\nAverage content age (days):")
print(df["content_age_days"].mean())

print("\nAverage days since last update:")
print(df["days_since_last_update"].mean())

Number of content pages: 30000

Trend direction counts:
                 Count  Percentage
trend_direction                   
down             16262       54.21
stable            5962       19.87
up                4388       14.63
new               2236        7.45
flat              1152        3.84

Average content age (days):
256.1678

Average days since last update:
46.0983


In [15]:
df[["ctr", "avg_position", "content_age_days"]].describe()

,ctr,avg_position,content_age_days
count,30000.000000,30000.00000,30000.00000
mean,0.510733,16.34238,256.16780
std,3.279162,15.21679,132.70793
min,0.000000,0.00000,90.00000
25%,0.000000,6.20000,132.00000
50%,0.070000,10.80000,236.00000
75%,0.290000,22.30000,333.00000
max,100.000000,245.00000,564.00000


The starter dataset contains 30,000 content pages. Initial exploration shows that 54.21% of pages are trending downward, suggesting that many pages may require review. The average content age is 256 days, indicating that many pages have existed long enough to potentially require updates. These observations suggest there is enough variation in freshness and performance to make content opportunity scoring a meaningful problem.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project can identify patterns in historical content performance and provide decision-support for prioritizing pages that may deserve review. The recommendations are based on observed relationships between features such as traffic trends, content age, CTR, ranking position, and engagement.

However, this work cannot prove that refreshing a page will improve rankings or traffic. It cannot predict Google's ranking algorithm or establish causal relationships. External factors such as seasonality, competition, algorithm updates, and changes in user behavior may influence performance independently of the model's recommendations.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.